### PR TITLE
Added support for reading template metadata from metadata.json #32

### LINKED ISSUE #32: Additionally use metadata.json content
If a `metadata.json` file exists in the same directory as the template then these values can be used.

Values in the template should take precedence over `metadata.json`.

[Reference](https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE#metadatajson)

### LINKED ISSUE #60: Add template summary
Azure Quickstart templates include an additional summary within `metadata.json`, however currently we don't support this metadata property.

Supporting an additional `summary` property would allow separation between a short synopsis and a longer description.

### PR DESCRIPTION
**Pull request by BernieWhite**
## PR Summary

- Added support for reading template metadata from `metadata.json`. #32
  - This adds additional compatibility for the Azure Quickstart templates repository.
  - Additional metadata from `metadata.json` will be read when it exists.
  - Template metadata take priority over `metadata.json`.
- Added support for the `summary` template metadata property. #60
  - The `summary` template metadata property is intended to provide a short description of the template.
  - Use the `description` template metadata property to provide a detailed description of the template.
  
Fixes #32 
Fixes #60

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**

### CODE REVIEW

File: docs/features.md [hunk 1]
```diff
@@ -35,3 +35,55 @@ PSDocs runs on MacOS, Linux and Windows.
 PowerShell makes it easy to integrate PSDocs into popular CI systems.
 To install, use the `Install-Module` cmdlet within PowerShell.
 For installation options see [install instructions](install-instructions.md).
+
+## Frequently Asked Questions (FAQ)
+
+### Can PSDocs read from metadata.json?
+
+The Azure Quickstart Templates repository uses an additional `metadata.json` to store template metadata.
+PSDocs doesn't require a `metadata.json` file to exist but will fallback to this file if it exists.
```
Comment (vicperdana) [inline docs/features.md:44]: Should we just mention that the `metadata.md` is case insensitive/sensitive?

Comment (vicperdana) [reply docs/features.md:44]: Also for other properties within metadata.json?

File: tests/PSDocs.Azure.Tests/template-test/template.json [hunk 2]
```diff
@@ -2,7 +2,6 @@
     "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
     "contentVersion": "1.0.0.0",
     "metadata": {
-        "name": "Storage Account",
```
Comment (vicperdana) [inline tests/PSDocs.Azure.Tests/template-test/template.json:5]: Did you want to show that this property will be overriden by "itemDisplayName" in metadata.json?

File: tests/PSDocs.Azure.Tests/template-test/metadata.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+    "itemDisplayName": "Storage Account",
```
Comment (vicperdana) [inline tests/PSDocs.Azure.Tests/template-test/metadata.json:3]: Perhaps this should be named slightly differently to show the effect of having metadata.json?

**Review by vicperdana — APPROVED** [review]
Just added some comments for considerations

File: docs/features.md (hunk 1 above)
Comment (BernieWhite) [reply docs/features.md:44]: File name would be on Linux. Properties of `metadata.json` aren't case sensitive.

File: tests/PSDocs.Azure.Tests/template-test/template.json (hunk 2 above)
Comment (BernieWhite) [reply tests/PSDocs.Azure.Tests/template-test/template.json:5]: Properties in the template will override metadata.json. Updated to make this clearer.

File: tests/PSDocs.Azure.Tests/template-test/metadata.json (hunk 3 above)
Comment (BernieWhite) [reply tests/PSDocs.Azure.Tests/template-test/metadata.json:3]: Thanks Vic. It's a predefined schema so the properties are what they are. I'll expand the mapping table to provide some additional detail, which should hopefully address it.

**Comment by BernieWhite** [comment]
@vicperdana Revised to make usage clearer and more closely align with Quickstart template metadata.

**Review by vicperdana — APPROVED** [review]

**Comment by vicperdana** [comment]
@BernieWhite looks good.

**Pull request merged**